core/src/main/java/com/github/mizool/core/exception/UnprocessableEntityException.java

```diff
@@ -1,9 +1,13 @@
 package com.github.mizool.core.exception;
 
 /**
- * Thrown when the entity is syntactically correct (e.g. all non-null properties are set), but could not be processed
- * for semantic reasons (e.g. invalid references to other database entities).
+ * @deprecated It is designed to be thrown when the entity is syntactically correct (e.g. all non-null properties are set),
```

Comment:
Not having a class description is bad style, and "sabotages" things like package overview tables in Javadoc and IDE tooltips.

Your deprecation comment is quite good, though. I just rephrased it a bit and incorporated the abstract class mentioned in my comment below.

> `@deprecated Throwing UnprocessableEntityException doesn't say <i>why</i> the entity is unprocessable and is generally a bad fit as no client ever really has a chance of handling such a response. Instead, consider creating an exception class specific to the respective problem that extends {@link AbstractUnprocessableEntityException}.`